File: rust/src/models/pagination_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PaginationResponse {
    pub page: i32,

    pub per_page: i32,

    pub total_items: i32,

    pub total_pages: i32,
}

impl PaginationResponse {
    pub fn new(page: i32, per_page: i32, total_items: i32, total_pages: i32) -> Self {
        Self {
            page,
            per_page,
            total_items,
            total_pages,
        }
    }
}

File: typescript/src/models/paginationResponse.ts
// this file is @generated

export interface PaginationResponse {
  page: number;

  perPage: number;

  totalItems: number;

  totalPages: number;
}

export const PaginationResponseSerializer = {
  _fromJsonObject(object: any): PaginationResponse {
    return {
      page: object["page"],
      perPage: object["per_page"],
      totalItems: object["total_items"],
      totalPages: object["total_pages"],
    };
  },

  _toJsonObject(self: PaginationResponse): any {
    return {
      page: self.page,
      per_page: self.perPage,
      total_items: self.totalItems,
      total_pages: self.totalPages,
    };
  },
};
